src/components/Filter.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Input from "./reusables/Input";
+import { Button } from "library-simplified-reusable-components";
+
+export interface FilterProps {
+  setFilter: (filter: string) => void;
+  filterKeys: { [key: string]: boolean };
+  flipFilter: () => void;
+  title?: string;
+  buttonText?: string[];
+  initialFlip?: boolean;
+}
+
+export interface FilterState {
+  flip: boolean;
+}
+
+export default class Filter extends React.Component<FilterProps, FilterState> {
+  constructor(props: FilterProps) {
+    super(props);
+    this.flip = this.flip.bind(this);
+    this.state = { flip: this.props.initialFlip || false };
+  }
+
+  applyFilter(e) {
+    this.props.setFilter(e.currentTarget.name);
+  }
+
+  flip() {
+    this.setState({ flip: !this.state.flip });
+    this.props.flipFilter();
+  }
+
+  render(): JSX.Element {
+    let [flipped, unflipped] = this.props.buttonText || ["DO NOT have", "have"];
+    let filterButton = <Button className="inline inverted top-align" content={this.state.flip ? flipped : unflipped} callback={this.flip} />;
+    let title = <h3>{this.props.title || "Show items which"} {filterButton}</h3>;
```

Comment:
I don't think there's an h1 or h2 on the page and maybe this doesn't need to be a heading.